docs/source/developingplugins.rst

```diff
@@ -0,0 +1,48 @@
+.. _developingpluginsref:
+
+Developing Plugins
+==================
+
+*SQLFluff* is extensible through "plugins". We use the `pluggy library`_
+to make linting Rules pluggable, which enable users to implement rules that
+are just too "organization specific" to be shared, or too platform specific
+to be included in the core library.
+
+.. _`pluggy library`: https://pluggy.readthedocs.io/en/latest/
+
+Creating a plugin
+-----------------
+
+We have an example plugin in `sqlfluff/plugins/sqlfluff-plugin-example`_
+which you can use as a template.
+
+Few things to note about plugins:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Right now, only Rules can be added through plugins.
+
+The name of a plugin should start with *"sqlfluff-plugin"* to be valid.
```

Comment:
Can one plugin package provide multiple rules?

Reply:
Yes! I should probably document it

Reply:
Cool! I originally suggested this feature, and I imagine my company (well, me -- ha!) will end up writing multiple rules, starting with the security idea I've mentioned a few times.